apps/lottie-viewer/src/locations/ConfigScreen.tsx

```diff
@@ -0,0 +1,147 @@
+import { ConfigAppSDK } from '@contentful/app-sdk';
+import { Heading, Paragraph, Autocomplete, Flex, Checkbox, Pill, TextLink, FormLabel, Card, Text, Note } from '@contentful/f36-components';
+import tokens from '@contentful/f36-tokens';
+import { useSDK } from '@contentful/react-apps-toolkit';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { styles } from './ConfigScreen.styles';
+import { ExternalLinkIcon } from '@contentful/f36-icons';
+import { useJsonFieldsState } from '@src/hooks/useJsonFieldsState';
+import { buildEditorInterfaceControls, getJsonFields, groupFieldsByContentType } from '@src/configUtils';
+
+const ConfigScreen = () => {
+  const sdk = useSDK<ConfigAppSDK>();
+  const [parameters, setParameters] = useState<Record<string, any>>({});
+  const [inputValue, setInputValue] = useState<string>('');
+  const installTriggeredRef = useRef<boolean>(false);
+
+  const { jsonFields, jsonFieldsRef, initialize, updateField, resetOriginalState } = useJsonFieldsState();
+
+  const items = useMemo(
+    () =>
+      jsonFields.map((field) => ({
+        name: `${field.contentTypeName} > ${field.fieldName}`,
+        id: field.fieldId,
+        isChecked: field.isEnabled,
+        contentTypeId: field.contentTypeId,
+      })),
+    [jsonFields, inputValue]
+  );
+
+  const onConfigure = useCallback(async () => {
+    installTriggeredRef.current = true;
+    return { parameters };
+  }, [parameters]);
+
+  useEffect(() => {
+    sdk.app.onConfigure(onConfigure);
+  }, [sdk, onConfigure]);
+
+  useEffect(() => {
+    sdk.app.onConfigurationCompleted(async () => {
```

Comment:
I had to add this because onConfigure kept overwriting the changes I'd make to update `editorInterface` so I have to watch for when the other  configuration steps complete and then send the updates. I have a ref to know whether this effect ran because the install was triggered. But this is where we're actually sending those batch updates through the CMA and then resetting the state.